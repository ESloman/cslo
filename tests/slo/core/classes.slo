
class TestClass {
    func __init__(name, age) {
        self.name = name;
        self.age = age;
    }

    func hello() {
        println("My name is: " + self.name);
    }
}

var test = TestClass("elliot", 29);
var h = test.hello;
println(h);
h();
println(test);
println(test.name);
println(test.age);
println(test.hello);
test.hello();

var test2 = TestClass("bob", 43);
test2.hello();
