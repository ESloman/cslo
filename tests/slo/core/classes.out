<fn hello>
My name is: elliot
TestClass instance
elliot
29
<fn hello>
My name is: elliot
My name is: bob
